src/expression.c

```diff
@@ -9809,6 +9810,69 @@ Expression *SliceExp::syntaxCopy()
     return se;
 }
 
+bool SliceExp::isOpDollar(VarExp* ve)
+{
+    return ve->var == this->lengthVar;
+    // VarDeclaration* decl = ve->var->isVarDeclaration();
+    // return decl && decl->ident == Id::dollar;
+}
+
+SliceExp::Boundness SliceExp::analyzeRelativeBound(Expression* e,
```

Comment:
This doesn't need to be a member function, and if it's private to this file you can also move `Boundness` out of the header.